go-controller/pkg/ovn/obj_retry.go

```diff
@@ -126,18 +189,1067 @@ func (r *retryObjs) getObjRetryEntry(key string) *retryObjEntry {
 	return nil
 }
 
-// triggerRetryObjs track the retry objects map and every 30 seconds check if any object need to be retried
-func (oc *Controller) triggerRetryObjs(retryChan chan struct{}, iterateObjs func(bool)) {
-	go func() {
-		for {
-			select {
-			case <-time.After(retryObjInterval):
-				iterateObjs(false)
-			case <-retryChan:
-				iterateObjs(true)
-			case <-oc.stopChan:
+var sep = "/"
+
+func splitNamespacedName(namespacedName string) (string, string) {
+	if strings.Contains(namespacedName, sep) {
+		s := strings.SplitN(namespacedName, sep, 2)
+		if len(s) == 2 {
+			return s[0], s[1]
+		}
+	}
+	return namespacedName, ""
+}
+
+func getNamespacedName(namespace, name string) string {
+	return namespace + sep + name
+}
+
+// hasResourceAnUpdateFunc returns true if the given resource type has a dedicated update function.
+// It returns false if, upon an update event on this resource type, we instead need to first delete the old
+// object and then add the new one.
+func hasResourceAnUpdateFunc(objType reflect.Type) bool {
+	switch objType {
+	case factory.PodType,
+		factory.NodeType,
```

Comment:
The node is deleted through `deleteNodeEvent` only if there is a retry entry marked for deletion (https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/ovn.go#L1134-L1144), otherwise we just call `addUpdateNodeEvent` with the right parameters packed in `nodeSyncs` (https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/ovn.go#L1148-L1163)

Reply:
right we still have logic in the watcher to handle updating a node instead of a blind delete then add